diff -rupN original/usr/share/jolla-calendar/cover/CalendarCover.qml patch/usr/share/jolla-calendar/cover/CalendarCover.qml
--- original/usr/share/jolla-calendar/cover/CalendarCover.qml	2017-02-14 09:35:54.000000000 +0000
+++ patch/usr/share/jolla-calendar/cover/CalendarCover.qml	2017-02-21 07:33:43.000000000 +0000
@@ -2,8 +2,18 @@ import QtQuick 2.0
 import Sailfish.Silica 1.0
 import org.nemomobile.time 1.0
 import org.nemomobile.calendar 1.0
+import org.nemomobile.configuration 1.0
 
 CoverBackground {
+    property int indexView: 0
+    ConfigurationGroup {
+        id: calSettings
+        path: "/apps/jolla-calendar/settings"
+        property int nDaysForward: 7
+        property int daysWeeksMonths: 0
+        property int leftAction: 0
+        property int rightAction: 1
+    }
     Label {
         //% "New event"
         text: qsTrId("calendar-la-new_event")
@@ -20,6 +30,7 @@ CoverBackground {
     }
 
     CoverActionList {
+        enabled: eventList.count <= 2
         CoverAction {
             iconSource: "image://theme/icon-cover-new"
             onTriggered: {
@@ -29,6 +40,49 @@ CoverBackground {
             }
         }
     }
+    CoverActionList {
+        enabled: eventList.count > 2 
+        CoverAction {
+
+        function doLeftAction ()
+        {
+                if (calSettings.leftAction ==0) {
+                    app.activate()
+                    app.showMainPage(PageStackAction.Immediate)
+                    pageStack.push("../pages/EventEditPage.qml", {}, PageStackAction.Immediate)
+                } else {
+                    indexView = indexView-2
+                    if(indexView < 0) {
+                        indexView = (eventList.count % 2 == 0) ? eventList.count - 2:eventList.count - 1
+                    }
+                    eventList.positionViewAtIndex(indexView, ListView.Beginning)
+                }
+                return
+            }
+            iconSource:(calSettings.leftAction ==0) ? "image://theme/icon-cover-new" : "image://theme/icon-cover-previous"
+                onTriggered: doLeftAction() 
+        }
+        CoverAction {
+        function doRightAction ()
+        {
+                if (calSettings.rightAction ==0) {
+                    app.activate()
+                    app.showMainPage(PageStackAction.Immediate)
+                    pageStack.push("../pages/EventEditPage.qml", {}, PageStackAction.Immediate)
+                } else {
+                    indexView = (indexView+2)
+                    if(indexView >= eventList.count) {
+                        indexView = 0
+                    }
+                    eventList.positionViewAtIndex(indexView, ListView.Beginning)
+                    }
+                return
+            }
+             iconSource:  (calSettings.rightAction ==0) ? "image://theme/icon-cover-new" : "image://theme/icon-cover-next"
+                onTriggered:  doRightAction()
+        }
+    }
+    
     Column {
         x: Theme.paddingLarge
         spacing: Theme.paddingSmall
@@ -87,6 +141,7 @@ CoverBackground {
                         if (event.allDay || now < occurrence.endTime) {
                             activeAndComing.append({ displayLabel: event.displayLabel, allDay: event.allDay,
                                                        startTime: occurrence.startTime, endTime: occurrence.endTime,
+evDay: occurrence.time,
                                                        color: event.color })
 
                             if (!event.allDay && (nextEnding == undefined || occurrence.endTime < nextEnding)) {
@@ -113,6 +168,29 @@ CoverBackground {
                 id: allEvents
                 startDate: wallClock.time
                 onUpdated: eventUpdater.update()
+                 
+                function getEndDate() {
+                   switch(calSettings.daysWeeksMonths) {
+                       case 0:{
+                           return  QtDate.addDays(wallClock.time, calSettings.nDaysForward)
+                       }
+                       case 1:{ 
+                           return QtDate.addDays(wallClock.time, calSettings.nDaysForward *7)
+                       }
+                       case 2:{
+                           var addM =calSettings.nDaysForward
+                           var addY = 0
+                           if (addM >12) {
+                              addM = addM - 12
+                              addY = 1
+                           }
+ 
+                           var endDate = new Date(wallClock.time.getFullYear()+addY, wallClock.time.getMonth() + addM, wallClock.time.getDate())
+                           return endDate
+                        }
+                    }
+                }
+                endDate: getEndDate() 
             }
 
             ListView {
@@ -124,8 +202,8 @@ CoverBackground {
                 model: activeAndComing
                 interactive: false
                 width: parent.width
-                height: 2*eventHeight + spacing
-                spacing: Theme.paddingSmall
+                height: (count - indexView) == 1 ? eventHeight + spacing : 2*eventHeight + 2*spacing
+                spacing: Theme.paddingSmall/2
                 visible: count > 0
 
                 delegate: CoverEventItem {
diff -rupN original/usr/share/jolla-calendar/cover/CoverEventItem.qml patch/usr/share/jolla-calendar/cover/CoverEventItem.qml
--- original/usr/share/jolla-calendar/cover/CoverEventItem.qml	2017-02-14 09:36:08.000000000 +0000
+++ patch/usr/share/jolla-calendar/cover/CoverEventItem.qml	2017-02-27 08:58:00.000000000 +0000
@@ -26,20 +26,81 @@ Row {
         id: labelColumn
         spacing: -Theme.paddingSmall
         anchors.verticalCenter: parent.verticalCenter
-        EventTimeLabel {
+        
+        Label{
+           id: label
+
+           function addSuffix(i,todayDate) 
+           {
+
+              if((i - todayDate) ==0 ) {
+                  return ("Today")
+              }
+              if((i - todayDate) ==1 ) {
+                  return ("Tomorrow")
+              }
+ 
+              var j = i % 10
+              var k = i % 100
+              if (j == 1 && k != 11) { return i + "st" } 
+              if (j == 2 && k != 12) { return i + "nd"} 
+              if (j == 3 && k != 13) { return i + "rd" } 
+              return i + "th" 
+           }
+
+           function fdw(sTime,eTime) {
+              var t = new Date
+              var t2 = new Date(sTime)
+              var t3 = new Date(eTime)
+              var today = new Date(t.getFullYear(), t.getMonth(), t.getDate())
+              var startDay = new Date(t2.getFullYear(), t2.getMonth(), t2.getDate())
+              var endDay = new Date(t3.getFullYear(), t3.getMonth(), t3.getDate())
+              var delta = (startDay - today) / 86400000
+              var delta2 = (endDay - startDay) / 86400000
+              var dLabel
+              if (delta == 0) {
+                  //% "Today"
+                  dLabel = qsTrId("calendar-today")
+              }else if (delta == 1) {
+                  //% "Tomorrow"
+                  dLabel = qsTrId("calendar-tomorrow")
+              } else {
+                  //: Long date pattern without year. Used e.g. in month view.
+                  //% "d MMMM"
+                  dLabel = capitalize(Qt.formatDate(sTime, qsTrId("calendar-date_pattern_date_month")))
+              }
+              if(delta2 != 0) {
+              dLabel = dLabel + " - "+ addSuffix(endDay.getDate(),today.getDate())
+              } 
+              return dLabel
+          }
+
+          function capitalize(string) {
+              return string.charAt(0).toUpperCase() + string.substr(1)
+          }
+
+
+          text: fdw(startTime,endTime)
+          opacity: 0.6
+          font.pixelSize: Theme.fontSizeSmall
+          color: Theme.primaryColor
+          height: root.height/3
+        }
+        
+        CovTimeLabel {
             id: timeLabel
             opacity: 0.6
             font.pixelSize: Theme.fontSizeSmall
             verticalAlignment: Text.AlignVCenter
             fontSizeMode: Text.VerticalFit
-            height: root.height/2
+            height: root.height/3
         }
         Label {
             id: nameLabel
             font.pixelSize: Theme.fontSizeSmall
             verticalAlignment: Text.AlignVCenter
             fontSizeMode: Text.VerticalFit
-            height: root.height/2
+            height: root.height/3
         }
     }
 }
